lib/repository/repository.dart

```diff
@@ -18,6 +21,104 @@ abstract class Repository {
+  /// Input: [key] the key against which the value is to be set, [value] the value that is to be set.
+  bool setCacheDynamicType({required String key, required dynamic value});
+
+  /// This method will return the saved String if exists
+  /// Input: [key] the key of the value
+  /// Output: [String] the value of the key
+  dynamic getCacheDynamicType({required String key});
+
+  /// This method will delete the value from the cache
+  /// Input: [key] the key of the value
+  /// Output: [value] will return the value that is just removed
+  dynamic deleteCacheDynamic({required String key});
+
+  /// This method will generate the cookbook id for the easel app
+  /// Output: [String] the id of the cookbook which will contains all the NFTs.
+  Future<String> autoGenerateCookbookId();
+
+  /// This method will save the username of the cookbook generator
+  /// Input: [username] the username of the user who created the cookbook
+  /// Output: [bool] returns whether the operation is successful or not
+  Future<bool> saveCookBookGeneratorUsername(String username);
+
+  /// This method will save the artist name
+  /// Input: [name] the name of the artist which the user want to save
+  /// Output: [bool] returns whether the operation is successful or not
+  Future<bool> saveArtistName(String name);
+
+  /// This method will get the artist name
+  /// Output: [String] returns whether the operation is successful or not
+  String getArtistName();
+
+  /// This method will get the already created cookbook from the local database
+  /// Output: [String] if the cookbook already exists return cookbook else return null
+  String? getCookbookId();
+
+  /// This method will get the username of the cookbook generator
+  /// Output: [String] returns whether the operation is successful or not
+  String getCookBookGeneratorUsername();
+
+  /// This method will generate easel Id for the NFT
+  /// Output: [String] the id of the NFT that is going to be added in the recipe
+  String autoGenerateEaselId();
+
+  /// This method will save the draft of the NFT
+  /// Input: [NFT] the draft that will will be saved in database
+  /// Output: [int] returns id of the inserted document
+  /// will return error in the form of failure
+  Future<Either<Failure, int>> saveNft(NFT draft);
+
+  /// This method will update draft in the local database from description Page
+  /// Input: [id] the id of the nft,
+  /// [String] the  name of the nft , [String] the  description of the nft
+  /// [String] the  creator name of the nft , [String] the page name of the Pageview
+  /// Output: [bool] returns whether the operation is successful or not
+  /// will return error in the form of failure
+  Future<Either<Failure, bool>> updateNftFromDescription(int id, String nftName, String nftDescription, String creatorName, String step);
+
+  /// This method will update draft in the local database from Pricing page
+  /// Input: [id] the id of the nft, [String] the  name of the nft ,
```

Comment:
There is no nam of the nft in the params. So remove it  . [tradePercentage] should be used instead of [String]